### ISSUE TITLE
podman info crashes as non-root

### ISSUE THREAD

**Issue by TheloTheGreat** [containers/podman#12743]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

When run as non-root, the command `podman info` crashes. When run via sudo, it does not crash:

```
thelo@Experience:~/proj/podman-scratch$ podman info
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/containers/podman/pkg/cgroups.getCgroupPathForCurrentProcess(0x0, 0x0, 0x0, 0x0)
	github.com/containers/podman/pkg/cgroups/cgroups.go:235 +0x365
github.com/containers/podman/pkg/cgroups.getAvailableControllers(0x0, 0x1, 0x0, 0xc000527598, 0x13e6aa5, 0x1ad68f8, 0x22)
	github.com/containers/podman/pkg/cgroups/cgroups.go:135 +0x2fa
github.com/containers/podman/pkg/cgroups.GetAvailableControllers(0x0, 0x1, 0x0, 0x0, 0x5, 0x4456a5, 0xc)
	github.com/containers/podman/pkg/cgroups/cgroups.go:185 +0x3e
github.com/containers/podman/libpod.(*Runtime).hostInfo(0xc0004c4fc0, 0x5, 0xc0000fdf25, 0x5)
	github.com/containers/podman/libpod/info.go:103 +0x174
github.com/containers/podman/libpod.(*Runtime).info(0xc0004c4fc0, 0x32, 0x27f50c0, 0x0)
	github.com/containers/podman/libpod/info.go:39 +0x16d
github.com/containers/podman/libpod.(*Runtime).Info(...)
	github.com/containers/podman/libpod/runtime.go:863
github.com/containers/podman/pkg/domain/infra/abi.(*ContainerEngine).Info(0xc000122588, 0x1d5ba28, 0xc000510b10, 0xc00052e180, 0xc000527d30, 0x9c6a65)
	github.com/containers/podman/pkg/domain/infra/abi/system.go:32 +0x32
github.com/containers/podman/cmd/podman/system.info(0x2738040, 0x27f2908, 0x0, 0x0, 0x0, 0x0)
	github.com/containers/podman/cmd/podman/system/info.go:76 +0x76
github.com/spf13/cobra.(*Command).execute(0x2738040, 0xc00003c050, 0x0, 0x0, 0x2738040, 0xc00003c050)
	github.com/spf13/cobra/command.go:852 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0x273afc0, 0xc0000400c8, 0x18a2ea0, 0x27f2908)
	github.com/spf13/cobra/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra/command.go:897
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	github.com/spf13/cobra/command.go:890
main.Execute()
	github.com/containers/podman/cmd/podman/root.go:90 +0xe7
main.main()
	github.com/containers/podman/cmd/podman/main.go:38 +0x92
thelo@Experience:~/proj/podman-scratch$ sudo podman info
host:
  arch: amd64
  buildahVersion: 1.21.3
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - rdma
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: 'conmon: /usr/bin/conmon'
    path: /usr/bin/conmon
    version: 'conmon version 2.0.25, commit: unknown'
  cpus: 12
  distribution:
    distribution: ubuntu
    version: "21.10"
  eventLogger: journald
  hostname: Experience
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.13.0-23-generic
  linkmode: dynamic
  memFree: 8343576576
  memTotal: 16110825472
  ociRuntime:
    name: runc
    package: 'runc: /usr/sbin/runc'
    path: /usr/sbin/runc
    version: |-
      runc version 1.0.1-0ubuntu2
      spec: 1.0.2-dev
      go: go1.16.5
      libseccomp: 2.5.1
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: true
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 6442446848
  swapTotal: 6442446848
  uptime: 4h 2m 39.88s (Approximately 0.17 days)
registries: {}
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 2
    paused: 0
    running: 0
    stopped: 2
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 2
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.2.1
  Built: 0
  BuiltTime: Wed Dec 31 19:00:00 1969
  GitCommit: ""
  GoVersion: go1.16.7
  OsArch: linux/amd64
  Version: 3.2.1
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
thelo@Experience:~/proj/podman-scratch$ podman version
Version:      3.2.1
API Version:  3.2.1
Go Version:   go1.16.7
Built:        Wed Dec 31 19:00:00 1969
OS/Arch:      linux/amd64

```

**Output of `podman info --debug`:**

```
thelo@Experience:~/proj/podman-scratch$ sudo podman info --debug
host:
  arch: amd64
  buildahVersion: 1.21.3
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - rdma
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: 'conmon: /usr/bin/conmon'
    path: /usr/bin/conmon
    version: 'conmon version 2.0.25, commit: unknown'
  cpus: 12
  distribution:
    distribution: ubuntu
    version: "21.10"
  eventLogger: journald
  hostname: Experience
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.13.0-23-generic
  linkmode: dynamic
  memFree: 8345616384
  memTotal: 16110825472
  ociRuntime:
    name: runc
    package: 'runc: /usr/sbin/runc'
    path: /usr/sbin/runc
    version: |-
      runc version 1.0.1-0ubuntu2
      spec: 1.0.2-dev
      go: go1.16.5
      libseccomp: 2.5.1
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: true
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 6442446848
  swapTotal: 6442446848
  uptime: 4h 4m 29.59s (Approximately 0.17 days)
registries: {}
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 2
    paused: 0
    running: 0
    stopped: 2
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 2
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.2.1
  Built: 0
  BuiltTime: Wed Dec 31 19:00:00 1969
  GitCommit: ""
  GoVersion: go1.16.7
  OsArch: linux/amd64
  Version: 3.2.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
thelo@Experience:~/proj/podman-scratch$ sudo apt list podman
Listing... Done
podman/impish,now 3.2.1+ds1-2ubuntu3 amd64 [installed]
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

No, this is on Podman 3.2.1 from the Ubuntu repositories. I can reinstall a latest-dev version of podman if I actually need to, but this issue should be very quick to confirm or infirm anyway.

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Physical desktop machine running Ubuntu 21.10.

**Comment by rhatdan** [comment]
Weird that we have not seen this before.  Might be something strange in your cgroup setup.
@giuseppe WDYT?

**Comment by Luap99** [comment]
This is already fixed in v3.4, see https://github.com/containers/podman/pull/11463

**Comment by TheloTheGreat** [comment]
Thanks, after building podman 4.0.0-dev directly from git today, I can confirm that that version no longer has the crash problem.
